src/gf.c

```diff
@@ -2379,6 +2369,24 @@ jl_code_instance_t *jl_method_compiled(jl_method_instance_t *mi, size_t world)
     return NULL;
 }
 
+// Opportunistic SOURCE_MODE_ABI cache lookup. Note that it's possible for us to miss a CodeInstance for which the
+// compiler has already deleted the source, but not yet set ->invoke.
+jl_code_instance_t *jl_method_inferred_with_abi(jl_method_instance_t *mi JL_PROPAGATES_ROOT, size_t world)
+{
+    jl_code_instance_t *codeinst = jl_atomic_load_relaxed(&mi->cache);
+    for (; codeinst; codeinst = jl_atomic_load_relaxed(&codeinst->next)) {
+        if (codeinst->owner != jl_nothing)
+            continue;
+
+        if (jl_atomic_load_relaxed(&codeinst->min_world) <= world && world <= jl_atomic_load_relaxed(&codeinst->max_world)) {
+            jl_value_t *code = jl_atomic_load_relaxed(&codeinst->inferred);
+            if (code && (code != jl_nothing || (jl_atomic_load_relaxed(&codeinst->invoke) != NULL)))
```

Comment:
Maybe we should query the JIT directly here, since it knows the remaining bit of the puzzle here (if something is queued for JIT, then it will be in the linfo_in_flight, but might not have an `invoke` for a long time after that)

Reply:
Ok, let me look at doing this for now.

Reply:
Decided to go back to the ->invoke thing I proposed earlier.